miden/src/cli/data.rs

```diff
@@ -72,18 +93,134 @@ impl InputFile {
         Ok(inputs)
     }
 
+    /// Parse advice provider data from the input file.
     pub fn parse_advice_provider(&self) -> Result<MemAdviceProvider, String> {
-        let stack = self
-            .advice_stack
+        let mut advice_inputs = AdviceInputs::default();
+
+        let stack = self.parse_advice_stack().map_err(|e| format!("advice stack error: {e}"))?;
+        advice_inputs = advice_inputs.with_stack_values(stack).map_err(|e| e.to_string())?;
+
+        if let Some(map) = self.parse_advice_map().map_err(|e| format!("advice map error: {e}"))? {
```

Comment:
We have a richer context here that might provide our error with more information. It could be `failed to parse advice provider: {e}`

Reply:
updated